delivery_postlogistics/models/stock_picking.py

```diff
@@ -0,0 +1,287 @@
+# Copyright 2013-2016 Camptocamp SA
+# License AGPL-3.0 or later (https://www.gnu.org/licenses/agpl).
+import base64
+from operator import attrgetter
+
+from odoo import _, api, exceptions, fields, models
+
+from ..postlogistics.web_service import PostlogisticsWebService
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    delivery_type = fields.Selection(related="carrier_id.delivery_type")
+    carrier_code = fields.Char(related="carrier_id.code")
+    # TODO: Change the name of this field
+    option_ids = fields.Many2many(
+        comodel_name="postlogistics.delivery.carrier.option", string="Options"
+    )
+
+    delivery_fixed_date = fields.Date(
```

Comment:
This should somehow be linked with the sales order commitment_date when set

Reply:
yep, it should be, added